docs/09_tutorials/10_nested_containers/index.md

```diff
@@ -0,0 +1,206 @@
+| pair | [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/nestcontn2a.cpp#L998) | [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/nestcontn2a.cpp#L1022) | [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/nestcontn2a.cpp#L1048) | [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/nestcontn2a.cpp#L1079) | [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/nestcontn2a.cpp#L1105)  | [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/tupletestkv.cpp#L242) |
+| tuple |  [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/tupletestkv.cpp#L327) |  [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/tupletestkv.cpp#L296) |  [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/tupletestkv.cpp#L363) |  [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/tupletestkv.cpp#L402) |  [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/tupletestkv.cpp#L434)  | [supported](https://github.com/EOSIO/eosio.cdt/blob/46cfbd2c49a98a425fb0cc39243c3813c62ebef3/tests/toolchain/build-pass/tupletestkv.cpp#L264) |
+
+The above matrix is **applicable to** both **eosio::multi_index** and the **value part of eosio::kv::map**.
+The **std::tuple is supported only in the new version of *eosio-cpp* compiler**, previous versions exhibit a bug related to parsing AST.
+
+For example:
+
+```cpp
+using my_map_t = eosio::kv::map<"people2kv"_n, int, person2kv>
+```
+
+defines an `eosio::kv::map` type, which has `int` as its key, and struct `person2kv` as its value, and field `set<vec_uint16> stv` of its value part struct `person2kv` is a set of vectors; according to the first row of above table, set of vectors is *supported*, and the cleos commands of the related push actions *setstv, prntstv* are documented in the *.sh* scripts of this project.
+
+### Additional Rules
+
+* The **key part** of `eosio::kv::map` can be primitive types, e.g. `std::string`, `eosio::name`, which are all supported by current ESOIO implementation
+* If the *key part* of `eosio::kv::map` is a self-defined struct, then the self-defined struct has to be wrapped by **CDT_REFLECT**, as shown in below, which can be found in *nestcontn2kv.cpp*:
+
+    ```cpp
+    using my_map_t4 = eosio::kv::map<"people2kv4"_n, mystructrefl, person2kv>
+    ```
+
```

Comment:
And some words here to tell testkvkeys.cpp has many more examples to show various types can be the key of kv::map including eosio built-in types, STL container and nested containers